pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,276 @@
+		Kind:       "AgentCluster",
+		APIVersion: "capi-provider.agent-install.openshift.io/v1alpha1",
+		Name:       i.localClusterNamespace + "-cluster-install",
+		UID:        agentClusterInstall.UID,
+	}
+	clusterDeployment.OwnerReferences = []metav1.OwnerReference{agentClusterInstallOwnerRef}
+	err := i.clusterImportOperations.CreateClusterDeployment(clusterDeployment)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		i.log.Errorf("could not create ClusterDeployment due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (i *LocalClusterImport) createNamespace(name string) error {
+	err := i.clusterImportOperations.CreateNamespace(name)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		i.log.Errorf("could not create Namespace due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (i *LocalClusterImport) ImportLocalCluster() error {
+
+	var errorList error
+
+	clusterVersion, err := i.clusterImportOperations.GetClusterVersion("version")
+	if err != nil {
+		i.log.Errorf("unable to find cluster version info due to error: %s", err.Error())
+		errorList = multierror.Append(nil, err)
+	}
+
+	release_image := ""
+	if clusterVersion != nil && clusterVersion.Status.History[0].State != configv1.CompletedUpdate {
+		message := "the release image in the cluster version is not ready yet, please wait for installation to complete and then restart the assisted-service"
+		i.log.Error(message)
+		errorList = multierror.Append(errorList, errors.New(message))
+	}
+
+	kubeConfigSecret, err := i.clusterImportOperations.GetSecret("openshift-kube-apiserver", "node-kubeconfigs")
+	if err != nil {
+		i.log.Errorf("unable to fetch local cluster kubeconfigs due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	pullSecret, err := i.clusterImportOperations.GetSecret("openshift-machine-api", "pull-secret")
+	if err != nil {
+		i.log.Errorf("unable to fetch pull secret due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	dns, err := i.clusterImportOperations.GetClusterDNS()
+	if err != nil {
+		i.log.Errorf("could not fetch DNS due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	numberOfControlPlaneNodes, err := i.clusterImportOperations.GetNumberOfControlPlaneNodes()
+	if err != nil {
+		i.log.Errorf("unable to determine the number of control plane nodes due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	// If we already have errors before we start writing things, then let's stop
+	// better to let the user fix the problems than have to delete things later.
+	if errorList != nil {
+		return errorList
+	}
+
+	release_image = clusterVersion.Status.History[0].Image
+
+	err = i.createClusterImageSet(release_image)
+	if err != nil {
+		errorList = multierror.Append(errorList, err)
+	}
+
+	err = i.createNamespace(i.localClusterNamespace)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
```

Comment:
minor comment - you already handling already exist error inside `createNamespace`

Reply:
I actually changed this, I need to be able to mock this so I moved the handling of this stuff into import_local_cluster